convert2rhel/actions/__init__.py

```
@@ -230,63 +285,199 @@ def get_actions(actions_path, prefix):
+        if successes is None:
+            successes = []
+        else:
+            successes = list(successes)
+
+        if failures is None:
+            failures = []
+        else:
+            failures = list(failures)
+
+        if skips is None:
+            skips = []
+        else:
+            skips = list(skips)
+
+        for action_class in resolve_action_order(self.actions):
+            # Decide if we need to skip because deps have failed
+            failed_deps = [d for d in action_class.dependencies if d not in successes]
+
+            action = action_class()
+
+            if failed_deps:
+                to_be = "was"
+                if len(failed_deps) > 1:
+                    to_be = "were"
+                message = "Skipped because %s %s not successful" % (format_sequence_as_message(failed_deps), to_be)
+
+                action.set_result(status="SKIP", error_id="SKIP", message=message)
+                skips.append(action)
+                continue
+
+            # Run the Action
+            try:
+                action.run()
+            except (Exception, SystemExit) as e:
+                # Uncaught exceptions are handled by constructing a generic
+                # failure message here that should be reported
+                message = (
+                    "Unhandled exception was caught: %s"
+                    "\nPlease file a bug at https://issues.redhat.com/ to have this"
+                    " fixed or a specific error message added."
+                    "\nTraceback: %s" % (e, traceback.format_exc())
+                )
+                action.set_result(status=STATUS_CODE["ERROR"], error_id="UNEXPECTED_ERROR", message=message)
+
+            # Categorize the results
+            if action.status <= STATUS_CODE["WARNING"]:
+                successes.append(action)
+
+            if action.status > STATUS_CODE["WARNING"]:
+                failures.append(action)
+
```

Not sure if that's necessary or not, but should we change this to make sure that anything that is `>= SKIP` is a failure, instead of the currrent behavior of `> WARNING`? 

The categorization is correct here, but, I think it would make more sense from the point of view of the reader if we use `SKIP` rather than `WARNING` here, especially because `WARNING` will be included in the success case. 

Not sure if you agree with that logic, let me know if that doesn't apply here! 

```suggestion
            # Categorize the results
            if action.status <= STATUS_CODE["WARNING"]:
                successes.append(action)

            if action.status > STATUS_CODE["SKIP"]:
                failures.append(action)

```

---

Your point makes sense to me, with my suggestion we will be losing anything that is between the ranges. 

I would presume that we have a fixed point defined that starts from `WARNING` since it is the first thing we check, for consistency, I would keep the `WARNING` for the second one too.